chia/full_node/mempool_check_conditions.py

```diff
@@ -107,16 +108,38 @@ def add_cond(
     conds[op].append(ConditionWithArgs(op, args))
 
 
+def unwrap(x: Optional[uint32]) -> uint32:
+    assert x is not None
+    return x
+
+
 def get_name_puzzle_conditions(
-    generator: BlockGenerator, max_cost: int, *, cost_per_byte: int, mempool_mode: bool
+    generator: BlockGenerator, max_cost: int, *, cost_per_byte: int, mempool_mode: bool, height: Optional[uint32] = None
```

Comment:
```suggestion
    generator: BlockGenerator, max_cost: int, *, cost_per_byte: int, mempool_mode: bool, height: uint32 = uint32(0)
```

Any reason to not just drop the `Optional` and default to `0`? I guess we can as well assume a zero height is invalid here. This way you would assert `> 0` and you could drop the `unwrap`.

Reply:
the reason is that I don't want it to be possible to call this function, not in mempool mode and also not specifying the height. 0 *is* a valid height, but it would work to default it to -1. However, that just seems like a poor-mans's optional.